docs/integrations/istio.md

```diff
@@ -0,0 +1,334 @@
+kubectl label namespace vm istio-injection=enabled;
+
+# Output:
+# namespace/default labeled
+# namespace/vm labeled
+```
+Applying the `istio-injection=enabled` label signals Istio to inject the sidecar proxy into every pod launched in these namespaces. 
+This allows your workloads to seamlessly join the service mesh without modifying pod specifications.
+
+Re-deploy services:
+```sh
+kubectl -n vm rollout restart deployment vm-operator;
+kubectl -n vm rollout status deployment vm-operator --watch=true;
+
+kubectl -n vm rollout restart deployment vmsingle-demo;
+kubectl -n vm rollout status deployment vmsingle-demo  --watch=true;
+
+kubectl -n vm rollout restart deployment vmagent-demo; 
+kubectl -n vm rollout status deployment vmagent-demo  --watch=true;
+
+kubectl -n default rollout restart deployment demo-app; 
+kubectl -n default rollout status deployment demo-app  --watch=true;
+
+# Output:
+# ...
+# deployment "vm-operator" successfully rolled out
+# ...
+# deployment "vmsingle-demo" successfully rolled out
+# ...
+# deployment "vmagent-demo" successfully rolled out
+# ...
+# deployment "demo-app" successfully rolled out
+```
+
+To verify that Istio sidecars have been properly injected into the pods, run:
+```sh
+kubectl get pods --all-namespaces  -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{range .spec.containers[*]}- {.name}{"\n"}{end}{"\n"}{end}'
+
+# Output:
+# demo-app-5b65b4d975-h47g6
+# - main
+# - istio-proxy
+#
+# istiod-7f898458c5-pq878
+# - discovery
+#
+# ...
+# vmsingle-demo-7f5dd5cfd9-pspj2
+# - vmsingle
+# - istio-proxy
+```
+At this point, pods in the `vm` and `default` namespaces have the istio-proxy injected.
+Pods in other namespaces do not.
+
+Let's check `VMAgent` istio-proxy logs to verify that it is scraping metrics from the demo app:
+```sh
+DEMO_APP_POD=$(kubectl get pod -n default -l "app.kubernetes.io/name=demo-app" -o jsonpath="{.items[0].metadata.name}");
+DEMO_APP_POD_IP=$(kubectl get pod -n default ${DEMO_APP_POD} -o jsonpath='{.status.podIP}');
+VMAGENT_POD_NAME=$(kubectl get pod -n vm -l "app.kubernetes.io/name=vmagent" -o jsonpath="{.items[0].metadata.name}");
+
+kubectl -n vm logs -f "${VMAGENT_POD_NAME}" -c istio-proxy 2>&1 | grep "${DEMO_APP_POD_IP}";
+
+# Output:
+# [2025-05-25T11:46:50.187Z] "GET /metrics HTTP/1.1" 200 - via_upstream - "-" 0 43 2 1 "-" "vm_promscrape" "c5e5b566-d179-9e29-b02f-2c20382ae353" "10.244.0.13:8080" "10.244.0.13:8080" PassthroughCluster 10.244.0.12:57118 10.244.0.13:8080 10.244.0.12:57114 - allow_any
+```
+You can also check out `VMAgent` UI the steps outlined in the [Quick Start Scraping](https://docs.victoriametrics.com/operator/quick-start/#scraping) section.
+
+## Partially strict policy
+
+By default, `VMAgent` sends scrape requests directly to pod IP:PORT.
+Istio doesn’t support `STRICT` mode for direct pod-to-pod traffic.
+Instead, Istio expects traffic to go through Kubernetes Services to ensure security and [identity checks](https://spiffe.io).
```

Comment:
is this a right link?

Reply:
Replaced with the link to the official Istio site.

---

This is the right one, istio identity model is based on SPiffe.

```
$./99-print-vmagent-spiffe-uri.sh 
++ kubectl get pod -n vm -l app.kubernetes.io/name=vmagent -o 'jsonpath={.items[0].metadata.name}'
+ VMAGENT_POD_NAME=vmagent-demo-5665864b96-blrt7
+ kubectl -n vm exec -it vmagent-demo-5665864b96-blrt7 -c istio-proxy -- openssl x509 -in /etc/istio-certs/cert-chain.pem -noout -text
+ grep -A 1 'Subject Alternative Name'
            X509v3 Subject Alternative Name: critical
                URI:spiffe://cluster.local/ns/vm/sa/vmagent-demo
```